zerver/tests/test_events.py

```diff
@@ -490,6 +491,11 @@ def normalize(state: dict[str, Any]) -> None:
                 # since events don't carry the relevant information.
                 # Fix the value just like server_timestamp.
                 state["presence_last_update_id"] = 0
+            if "realm_topics_policy" in state:
+                state["realm_mandatory_topics"] = (
+                    self.user_profile.realm.topics_policy
+                    == RealmTopicsPolicyEnum.disable_empty_topic.value
+                )
```

Comment:
Why is this here? I think every block in this function should have a comment, but I also would expect code here to be papering over the API code itself being wrong.